lib/livebook/runtime.ex

```diff
@@ -127,19 +177,20 @@ defprotocol Livebook.Runtime do
   def drop_container(runtime, container_ref)
 
   @doc """
-  Asynchronously finds completion items matching the given `hint` text.
+  Asynchronously handles an intellisense request.
+
+  This part of runtime functionality is used to provide
+  language and context specific intellisense features in
+  the text editor.
 
-  The given `locator` idenfities an evaluation, which bindings
-  and environment are used to provide a more relevant completion
-  results. If there's no appropriate evaluation, `nil` refs can
-  be provided.
+  The response is sent to the `send_to` process as
+  `{:intellisense_response, ref, response}`.
 
-  Completion response is sent to the `send_to` process as
-  `{:completion_response, ref, items}`, where `items` is a
-  list of `t:Livebook.Runtime.completion_item/0`.
+  The given `locator` idenfities an evaluation that may be used
+  as context when resolving the request (if relevant).
   """
-  @spec request_completion_items(t(), pid(), term(), String.t(), locator()) :: :ok
-  def request_completion_items(runtime, send_to, completion_ref, hint, locator)
+  @spec handle_intellisense(t(), pid(), reference(), intellisense_request(), locator()) :: :ok
+  def handle_intellisense(runtime, send_to, ref, request, locator)
```

Comment:
I took this as an opportunity to unify the editor related functionalities (completion, hover, formatting) into this single runtime call, adhering to the request/response specification above. This saves quite a lot of duplication. It's kinda like a tiny language server now :)